design/Geth_type_conventions.md

```diff
@@ -0,0 +1,48 @@
+# Code Conventions Proposal - Geth Type Renames
+
+This proposal seeks to establish a standard on how the enclave codebase should use the standard `geth` types like `types.Transaction`, `types.Block` and the likes.
+Currently the code base sparsely uses renamed types like the ones defined in `common/types.go`. The suggestion of this proposal is that such type renames should be the standard
+and not the exception. 
+
+## Example and thought process behind the suggestion
+
+Lets take a look at an example function from the code:
+```go
+// SubmitBlock is used to update the enclave with an additional L1 block.
+func (rc *RollupChain) SubmitBlock(block types.Block, isLatest bool) (*common.BlockSubmissionResponse, error) {
+```
+
+Notice how the comment explains what type of block should be passed to it. However the same information can be conveyed without having a comment at all:
+
+```go
+func (rc *RollupChain) SubmitBlock(block common.L1.Block, isLatest bool) (*common.BlockSubmissionResponse, error) {
```

Comment:
how can you alias to "common.L1.Block"?

Reply:
I'm not entirely sure you can, but I think the auto generated solidity go bindings were doing subpackages. I have to look into it but i imagine it is /common/L1/types.go where we have defined Block = gethtypes.Block.
Should certainly be possible, but even if it has to be common.L1Block it is still fine.